src/main/MQ2ImGuiConsole.cpp

```diff
@@ -593,19 +599,15 @@ class ZepMode_ImGuiConsole : public Zep::ZepMode_Standard
 //----------------------------------------------------------------------------
 
 // This is the imgui container for the Zep component.
-struct ImGuiZepConsole : public mq::imgui::ConsoleWidget, public mq::imgui::ImGuiZepEditor
+struct ImGuiZepConsole : public mq::imgui::ConsoleWidget
```

Comment:
why change the inheritance chain like this? the design before was mixins, and now it's not -- what effect does that have on the current console?

Reply:
Because the zep editor itself is a lua exposable object there was some overlap with the interfaces. Now console is just a specialized version of zepeditor

Reply:
I would still much prefer the pattern of keeping this as separate mixin modules -- what exactly is the overlap?

Reply:
ImGuiZepEditor is itself an interface to lua which is what this PR exposes.
ConsoleWidget is a virtual interface for ImGuiZepConsole to be exposed to lua

so both ImGuiZepEditor and ImGuiZepConsole are lua exposure APIs

Then ImGuiZepConsole is derived from both of them. So if it polymorph you get weird overlaps they they both have a Clear() function etc.

Long term @brainiac will have to refactor ImGuiZepConsole to not be a Console Widget and only extent ImGuiZepEditor but that is out of the scope of this PR.

Reply:
No, that's not quite it. The ConsoleWidget is the exposed piece, not anything else. This PR needs to follow the same pattern by putting a TextWidget into include/mq/imgui and exposing _that_ to plugins. Lua is a plug-in, it shouldn't be directly including stuff from main. You need to be following the same design we already had in place for console to decouple this from the main code.

Reply:
I will let @brainiac comment, but that is how I initially had it setup and he explicitly wanted to remove the additional abstraction.

Reply:
I don't agree that we should be tightly coupling _anything_ in src/main with any plugins, we've specifically worked to decouple these things. It would be better to put it into the actual plugin API, but that's a lot more work than simply following the existing pattern, and keeps the door open to more easily refactor it to be in the API. Zep _should not_ be this transparent to plugins, it's an implementation detail.